src/bin/mir2wasm.rs

```diff
@@ -11,39 +10,75 @@ extern crate rustc_driver;
 #[link_args = "-lstdc++ -static-libstdc++"]
 extern { }
 
-use mir2wasm::trans;
+use mir2wasm::trans::{self, WasmTransOptions};
 use rustc::session::Session;
 use rustc_driver::{driver, CompilerCalls};
 use std::process;
 
-struct MiriCompilerCalls;
+struct WasmCompilerCalls {
+    options: WasmTransOptions
+}
+
+impl WasmCompilerCalls {
+    fn new(options: WasmTransOptions) -> WasmCompilerCalls {
+        WasmCompilerCalls {
+            options: options
+        }
+    }
+}
 
-impl<'a> CompilerCalls<'a> for MiriCompilerCalls {
+impl<'a> CompilerCalls<'a> for WasmCompilerCalls {
     fn build_controller(
         &mut self,
         _: &Session,
         _: &getopts::Matches
     ) -> driver::CompileController<'a> {
         let mut control = driver::CompileController::basic();
+        let options = self.options;
 
         control.after_analysis.stop = rustc_driver::Compilation::Stop;
-        control.after_analysis.callback = Box::new(|state| {
+        control.after_analysis.callback = Box::new(move |state| {
             state.session.abort_if_errors();
-            trans::trans_crate(&state.tcx.unwrap(), state.mir_map.unwrap())
+
+            let entry_fn = state.session.entry_fn.borrow();
+            let entry_fn = if let Some((node_id, _)) = *entry_fn { Some(node_id) } else { None };
+            trans::trans_crate(&state.tcx.unwrap(), state.mir_map.unwrap(), entry_fn, &options)
                 .unwrap(); // FIXME
         });
 
         control
     }
 }
 
-#[miri_run]
 fn main() {
     env_logger::init().unwrap();
 
-    let mut args: Vec<String> = std::env::args().collect();
-    args.push("--target=arm-unknown-linux-gnueabi".to_string());
-    match rustc_driver::run_compiler(&args, &mut MiriCompilerCalls) {
+    let wasm_compiler_args = ["--run", "-O", "-q", "-h", "--help"];
+    let rustc_args : Vec<String> = std::env::args().filter(|arg| !wasm_compiler_args.contains(&arg.as_ref())).collect();
+
+    // TODO: use a command line parsing library
+    let mut options = WasmTransOptions::new();
+    for flag in std::env::args().filter(|arg| wasm_compiler_args.contains(&arg.as_ref())) {
+        match flag.as_ref() {
+            "--run" => { options.interpret = true }
+            "-O" => { options.optimize = true }
+            "-q" => { options.print = false }
+            "-h" | "--help" => {
+                let usage = "mir2wasm [OPTIONS] INPUT \n\n\
+                             Options: \n    \
+                             -h, --help    Display this message \n    \
+                             -O            Optimize the compiled wast module \n    \
+                             --run         Run the compiled module through the interpreter, without printing it \n    \
```

Comment:
sure I'll change it (and commit once, possibly squashing it if you guys prefer, with all the changes for this review) 
It doesn't exaclty silence stdout wholesale, it silences the compiler's stdout but not the interpreter's